lib/services/item_manager.ts

```diff
@@ -241,6 +241,7 @@ export class ItemManager extends PureService {
     for (const item of discardedItems) {
       this.collection.discard(item);
     }
+    this.notesCollection.needsRebuilding = true;
```

Comment:
I feel like the collection itself should be smart enough to know when it needs rebuilding based on when its own properties change, rather than an external source telling it to rebuild.

Reply:
I ruled out subclassing as I explained in another comment. Maybe I do need to rename this class to avoid the confusion, so that an external source telling it to rebuild wouldn't seem strange.